src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -9225,20 +9225,32 @@ std::string const& ObjectMgr::GetScriptName(uint32 id) const
     return (id < _scriptNamesStore.size()) ? _scriptNamesStore[id] : empty;
 }
 
-uint32 ObjectMgr::GetScriptId(std::string const& name)
+uint32 ObjectMgr::GetScriptIdOrAdd(std::string const& name)
 {
     // use binary search to find the script name in the sorted vector
     // assume "" is the first element
     if (name.empty())
         return 0;
 
-    ScriptNameContainer::const_iterator itr = std::lower_bound(_scriptNamesStore.begin(), _scriptNamesStore.end(), name);
+    ScriptNameContainer::const_iterator itr = std::find(_scriptNamesStore.begin(), _scriptNamesStore.end(), name);
     if (itr == _scriptNamesStore.end() || *itr != name)
-        return 0;
+    {
+        _scriptNamesStore.push_back(name);
+        return _scriptNamesStore.size() - 1;
+    }
 
     return uint32(itr - _scriptNamesStore.begin());
 }
 
+bool ObjectMgr::FindScriptId(std::string const& name) const
+{
+    if (name.empty())
+        return 0;
+
+    ScriptNameContainer::const_iterator itr = std::find(_scriptNamesStore.begin(), _scriptNamesStore.end(), name);
+    return itr != _scriptNamesStore.end() && *itr == name;
```

Comment:
How come you are checking twice for equality ?

Reply:
Because this is a copypaste from GetScriptId done without understanding how it works

Also a function named FindScriptId should return something more than just a bool "does it exist"